### ISSUE TITLE
False positive for `Lint/NestedPercentLiteral` cop

### ISSUE THREAD

**Issue by yensaki** [bbatsov/rubocop#5314]
`Lint/NestedPercentLiteral` cop detects the code like this.
this code has an array with elements which contains `%+`.

```ruby
puts %w[1%+ 2]
```
I think that it is probably false positive.

--------

## Expected behavior

No offence of the `Lint/NestedPercentLiteral` cop

## Actual behavior

```
offensive.rb:3:6: W: Lint/NestedPercentLiteral: Within percent literals, nested percent literals do not function and may be unwanted in the result.
puts %w[1%+ 2]
     ^^^^^^^^^
```

## Steps to reproduce the problem

Array with elements which contains `%+`.

```
$ cat offensive.rb
# frozen_string_literal: true

puts %w[1%+ 2]
```

Run rubocop v0.52.0

```
$ rubocop offensive.rb
Inspecting 1 file
W

Offenses:

offensive.rb:3:6: W: Lint/NestedPercentLiteral: Within percent literals, nested percent literals do not function and may be unwanted in the result.
puts %w[1%+ 2]
     ^^^^^^^^^

1 file inspected, 1 offense detected
```  

This seems to be detected by `/#{percent_literal}\W/`.

## RuboCop version

```
$ rubocop -V
0.52.0 (using Parser 2.4.0.2, running on ruby 2.4.2 x86_64-darwin16)
```

**Comment by asherkach** [comment]
This bug is the result of [this regex](https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/lint/nested_percent_literal.rb#L27-L29) not being tight enough. Fix forthcoming...
